### PR TITLE
update submodule and related changes

### PR DESCRIPTION
**Pull request by davidbloss**
## Changelog

`JSONString` type switched to `JSON` type in `opslevel-go`. Applying update here

- [X] List your changes here
- [X] Make a `changie` entry

### CODE REVIEW

File: opslevel/resource_opslevel_property_definition.go
Comment (taimoor-at-opslevel) [inline opslevel/resource_opslevel_property_definition.go:42]: Calling `NewJSON` would cause the `jsonencoded()` input to be double encoded, resulting in an unusable `schema`, AFAIK

**Review by taimoor-at-opslevel — CHANGES_REQUESTED** [review]
We should pause and wait until the current terraform PRs are merged before adding another dependent PR to the queue, since there can be conflicting changes

1 https://github.com/OpsLevel/terraform-provider-opslevel/pull/175
2 https://github.com/OpsLevel/terraform-provider-opslevel/pull/174

File: opslevel/resource_opslevel_property_definition.go
Comment (rocktavious) [reply opslevel/resource_opslevel_property_definition.go:42]: we should definitely test this out but i'm pretty sure this is correct as is.  `NewJSON` calls `unmarshal` on the string input so its turning a "string" into a `map[string]any` but as type `opslevel.JSON`. Which is what we want here.

Comment (taimoor-at-opslevel) [reply opslevel/resource_opslevel_property_definition.go:42]: @rocktavious and @davidbloss you are both right, I was mistaken here since I confused `PropertyDefinition.Schema` and `Property.Value`. 

[The patch for property definitions in terraform is correctly using `opslevel.NewJSON()`.](https://github.com/OpsLevel/terraform-provider-opslevel/blob/8698a4c943741ce9dd3691c001c0e962dff8ab8e/opslevel/resource_opslevel_property_definition.go#L55)

**Comment by davidbloss** [comment]
Fixed elsewhere

**Pull request closed without merging**
